### PR TITLE
add --print-only feature

### PR DESCRIPTION
**Pull request by lverma14**
Functionality to print the changed code to stdout. Doesn't modify the actual code.
Current gopatch -p functionality completely modifies the target file and
outputs nothing to the command line. There is no option which could show us what the
modified code after applying the patch would look like which also doesn't actually modify the code.

This pull request extends gopatch to display changed code to stdout. This would give more 
context and can be used for integrating gopatch into arclint.

Now call to --print-only flag is associated with printing out the modified code to stdout,
if the is patch applied. It also prints the description comments from the applied patch to the stderr.

Components Changed:
main.go: added --print-onlyflag and its functionality

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ type options struct {
 	Patches        []string  `short:"p" long:"patch" value-name:"file"`
 	Diff           bool      `short:"d" long:"diff"`
 	DisplayVersion bool      `long:"version"`
+	Format         bool      `long:"stdout"`
```
Comment (abhinav) [inline main.go:63]: This field shouldn't be named Format.
It should be named "Stdout".

Although now that I think about it, both, the flag should be `--print` or `--print-only`?
```suggestion
	Print          bool      `long:"print-only"`
```

File: main.go [hunk 2]
```diff
@@ -87,6 +88,8 @@ func newArgParser() (*flags.Parser, *options) {
 	parser.FindOptionByLongName("diff").Description =
 		"Print a diff of the proposed changes to stdout but don't modify any files."
 
+	parser.FindOptionByLongName("stdout").Description =
+		"If true, don't modify the code and print only the formatted code to stdout."
```
Comment (abhinav) [inline main.go:92]:
```suggestion
	parser.FindOptionByLongName("print-only").Description =
		"If true, don't modify the code and print the changed code to stdout."
```

File: main.go [hunk 3]
```diff
@@ -287,9 +290,12 @@ func run(args []string, stdin io.Reader, stderr io.Writer, stdout io.Writer) err
 			continue
 		}
 
-		if opts.Diff {
+		switch {
+		case opts.Diff:
 			err = preview(filename, content, bs, comments, stderr, stdout)
-		} else {
+		case opts.Format:
```
Comment (abhinav) [inline main.go:296]:
```suggestion
		case opts.Print:
```

**Review by abhinav — COMMENTED** [review]
Please add a test verifying that this functionality works as desired.

File: e2e_test.go [hunk 4]
```diff
@@ -139,13 +140,65 @@ func runIntegrationTest(t *testing.T, testFile string) {
 		})
 	}
 }
+func runIntegrationPrintOnlyTest(t *testing.T, testFile string) {
```
Comment (mindaugasrukas) [inline e2e_test.go:143]: nit: add a space between functions for readability.

File: e2e_test.go [hunk 5]
```diff
@@ -139,13 +140,65 @@ func runIntegrationTest(t *testing.T, testFile string) {
...
+			if skipTest(testsPatchNotApply, testFile, tt.Name) {
+				assert.Equal(t, "", stdout.String())
+			} else {
+				assert.Equal(t, string(tt.Want), stdout.String())
+				assert.Equal(t, string(tt.WantComment), stderr.String())
+			}
+		})
+	}
+}
+func skipTest(testMap map[string]struct{}, testFile, testName string) bool {
```
Comment (mindaugasrukas) [inline e2e_test.go:190]: nit: add a space between functions for readability.

File: main.go [hunk 6]
```diff
@@ -200,15 +203,18 @@ func findFiles(patterns []string) (_ []string, err error) {
 	return sortedFiles, err
 }
 
+func printComments(comments []string, stderr io.Writer) {
+	for _, c := range comments {
+		fmt.Fprintln(stderr, c)
+	}
+}
 func preview(
```
Comment (mindaugasrukas) [inline main.go:211]: nit: add a space between functions for readability.

File: main_test.go [hunk 7]
```diff
@@ -162,3 +162,44 @@ func TestPreview(t *testing.T) {
 		assert.Equal(t, expectedString, outputString)
 	})
 }
+func TestPrintComments(t *testing.T) {
```
Comment (mindaugasrukas) [inline main_test.go:165]: nit: add a space between functions for readability.

**Review by mindaugasrukas — APPROVED** [review]
LGTM

**Review by abhinav — APPROVED** [review]

File: main.go [hunk 8]
```diff
@@ -262,7 +270,13 @@ func run(args []string, stdin io.Reader, stderr io.Writer, stdout io.Writer) err
 
 		f, comments, ok := patchRunner.Apply(filename, f)
 		// If at least one patch didn't match, there's nothing to do.
+		// If --print was passed, print the contents out as-is.
```
Comment (mindaugasrukas) [inline main.go:273]: please update the comment as there is no option `--print` but `--print-only`

**Comment by abhinav** [comment]
Oops, accidentally closed. Re-opening.

**Pull request closed without merging**